app/ui/setup.py

```diff
@@ -64,36 +64,27 @@ def setup_block():
     external_func_url = get_api_gateway_url(sf_region_without_public, sd_deployment)
     connection.Connection.get().call("INTERNAL.SETUP_EF_URL", external_func_url)
 
+    privileges = [
+        "EXECUTE MANAGED TASK",
+        "EXECUTE TASK",
+        "MANAGE WAREHOUSES",
+        "IMPORTED PRIVILEGES ON SNOWFLAKE DB",
+    ]
+    missing_privileges = perms.get_missing_account_privileges(privileges)
+    if len(missing_privileges) > 0:
+        perms.request_account_privileges(missing_privileges)
```

Comment:
This call is loading the permissions modal element and when the user either exits that modal (by granting permission or cancelling the window), will the `Setup` page be rendered again?

I am wondering if we need to have some call/button to "refresh" the state. I assume we're reloading the page since you don't have any logic after `request_account_privileges` but wanted to double check.

Reply:
great, thanks.